kernel/comps/keyboard/src/lib.rs

```diff
@@ -0,0 +1,353 @@
+    Dc2,          // Ctrl + R, device control 2
+    Dc3,          // Ctrl + S, device control 3
+    Dc4,          // Ctrl + T, device control 4
+    Nak,          // Ctrl + U, negative acknowledge
+    Syn,          // Ctrl + V, synchronous idle
+    Etb,          // Ctrl + W, end of trans. block
+    Can,          // Ctrl + X, cancel
+    Em,           // Ctrl + Y, end of medium
+    Sub,          // Ctrl + Z, substitute
+    Esc,          // Ctrl + [, escape
+    Fs,           // Ctrl + \, file separator
+    Gs,           // Ctrl + ], group separator
+    Rs,           // Ctrl + ^, record separator
+    Us,           // Ctrl + _, unit separator
+    Space,        // ' '
+    Exclamation,  // '!'
+    DoubleQuote,  // '"'
+    Hash,         // '#'
+    Dollar,       // '$'
+    Percent,      // '%'
+    Ampersand,    // '&'
+    SingleQuote,  // '''
+    LeftParen,    // '('
+    RightParen,   // ')'
+    Asterisk,     // '*'
+    Plus,         // '+'
+    Comma,        // ','
+    Minus,        // '-'
+    Period,       // '.'
+    ForwardSlash, // '/'
+    Zero,
+    One,
+    Two,
+    Three,
+    Four,
+    Five,
+    Six,
+    Seven,
+    Eight,
+    Nine,
+    Colon,       // ':'
+    SemiColon,   // ';'
+    LessThan,    // '<'
+    Equal,       // '='
+    GreaterThan, // '>'
+    Question,    // '?'
+    At,          // '@'
+    UppercaseA,
+    UppercaseB,
+    UppercaseC,
+    UppercaseD,
+    UppercaseE,
+    UppercaseF,
+    UppercaseG,
+    UppercaseH,
+    UppercaseI,
+    UppercaseJ,
+    UppercaseK,
+    UppercaseL,
+    UppercaseM,
+    UppercaseN,
+    UppercaseO,
+    UppercaseP,
+    UppercaseQ,
+    UppercaseR,
+    UppercaseS,
+    UppercaseT,
+    UppercaseU,
+    UppercaseV,
+    UppercaseW,
+    UppercaseX,
+    UppercaseY,
+    UppercaseZ,
+    LeftBracket,  // '['
+    BackSlash,    // '\'
+    RightBracket, // ']'
+    Caret,        // '^'
+    Underscore,   // '_'
+    Backtick,     // '`'
+    LowercaseA,
+    LowercaseB,
+    LowercaseC,
+    LowercaseD,
+    LowercaseE,
+    LowercaseF,
+    LowercaseG,
+    LowercaseH,
+    LowercaseI,
+    LowercaseJ,
+    LowercaseK,
+    LowercaseL,
+    LowercaseM,
+    LowercaseN,
+    LowercaseO,
+    LowercaseP,
+    LowercaseQ,
+    LowercaseR,
+    LowercaseS,
+    LowercaseT,
+    LowercaseU,
+    LowercaseV,
+    LowercaseW,
+    LowercaseX,
+    LowercaseY,
+    LowercaseZ,
+    LeftBrace,  // '{'
+    Pipe,       // '|'
+    RightBrace, // '}'
+    Tilde,      // '~'
+    Del,
+    UpArrow,
+    DownArrow,
+    RightArrow,
+    LeftArrow,
+    End,
+    Home,
+    Insert,
+    Delete,
+    PageUp,
+    PageDown,
+    F1,
+    F2,
+    F3,
+    F4,
+    F5,
+    F6,
+    F7,
+    F8,
+    F9,
+    F10,
+    F11,
+    F12,
+}
+
+impl InputKey {
+    /// Gets the xterm control sequences for this key.
+    ///
+    /// Reference: <https://invisible-island.net/xterm/ctlseqs/ctlseqs.pdf>
+    pub fn as_xterm_control_sequences(&self) -> &[u8] {
```

Comment:
I guess for one key, there is a control sequence that contains multiple characters, right?

If so, maybe:
```patch
-    /// Gets the xterm control sequences for this key.
+    /// Gets the xterm control sequence for this key.
     ///
     /// Reference: <https://invisible-island.net/xterm/ctlseqs/ctlseqs.pdf>
-    pub fn as_xterm_control_sequences(&self) -> &[u8] {
+    pub fn as_xterm_control_sequence(&self) -> &[u8] {
```